koptional/src/test/kotlin/com/gojuno/koptional/OptionalSpec.kt

```diff
@@ -83,4 +83,25 @@ class OptionalSpec : Spek({
 
         }
     }
+
+    describe("component1") {
```

Comment:
Let's add few tests to check type system behavior?

I was afraid that this PR would break previous construction:

```kotlin
val (value: String) = Some("str")
```

Because now `Optional` declares `component1()` as `T?`.

But gladly compiler actually generates proper function override in `Some` with `@NotNull`:

```
class Some extends Optional {
   @NotNull
   public final Object component1() {
      return this.value;
   }
```

So we get both `T?` if you destructure `Optional` and `T` if you destructure `Some<T>`, but I'm afraid that this behavior can be easily broken in compiler and that there are not so many similar use cases to report that to them quickly

Reply:
Something like:

```kotlin
context("destructure Some") {

    val some = Some("string")

    it("destructures to non-null type") {
        val (value: String) = some
    }
}
```

But for all types: `Optional`, `Some`, `None`. Right now you test behavior but not the type information